bitnami/etcd/3.5/debian-11/rootfs/opt/bitnami/scripts/libetcd.sh

```diff
@@ -664,6 +664,11 @@ etcd_initialize() {
             if is_boolean_yes "$ETCD_DISABLE_PRESTOP"; then
                 info "The member will try to join the cluster by it's own"
                 export ETCD_INITIAL_CLUSTER_STATE=existing
+                #
+                # If ETCD_DISABLE_PRESTOP is set, we won't dynamically adjust membership. In this case
+                # we'll return and allow etcd to start and join the statically configured cluster.
+                #
+                return 0
             fi
             member_id="$(get_member_id)"
```

Comment:
I'm worried that skipping all the cluster join/rejoin if ETCD_DISABLE_PRESTOP is enabled. It would skip the logic where 'initial-cluster-state' is being added to etcd configuration:

```
[[ -f "$ETCD_CONF_FILE" ]] && etcd_conf_write "initial-cluster-state" "$ETCD_INITIAL_CLUSTER_STATE"
```
Why not add it as an additional condition below?

```suggestion
            member_id="$(get_member_id)"
            if is_boolean_yes "$ETCD_DISABLE_PRESTOP"; then
                info "The member will try to join the cluster by it's own"
                export ETCD_INITIAL_CLUSTER_STATE=existing
            elif ! is_healthy_etcd_cluster; then
            warn "Cluster not responding!"
```

In that way it won't accidentally skip any logic that may be added at the end of etcd_initialize.

Reply:
@mdhont great suggestion, i like that much better.  tested with this fix and things look good:

```
etcd 14:48:02.38 INFO  ==> ** Starting etcd setup **
etcd 14:48:02.44 INFO  ==> Validating settings in ETCD_* env vars..
etcd 14:48:02.45 WARN  ==> You set the environment variable ALLOW_NONE_AUTHENTICATION=yes. For safety reasons, do not use this flag in a production environment.
etcd 14:48:02.48 INFO  ==> Initializing etcd
etcd 14:48:02.49 INFO  ==> Generating etcd config file using env variables
etcd 14:48:02.68 INFO  ==> Detected data from previous deployments
etcd 14:48:02.72 DEBUG ==> Setting data directory permissions to 700 in a recursive way (required in etcd >=3.4.10)
etcd 14:48:02.74 INFO  ==> The member will try to join the cluster by it's own
etcd 14:48:03.05 DEBUG ==> cray-etcd-test-bitnami-etcd-0.cray-etcd-test-bitnami-etcd-headless.services.svc.cluster.local:2379 endpoint is active
etcd 14:48:03.24 DEBUG ==> cray-etcd-test-bitnami-etcd-2.cray-etcd-test-bitnami-etcd-headless.services.svc.cluster.local:2379 endpoint is active
etcd 14:48:03.32 INFO  ==> Updating member in existing cluster
Member 6cf1858af3bb04d9 updated in cluster  dbffc1abc62b48a
etcd 14:48:03.60 INFO  ==> ** etcd setup finished! **
```

Reply:
pushed a commit with suggested change..

Reply:
Great! Could you sign the commit:

https://github.com/bitnami/containers/pull/38514/checks

Then I think we can land it.

Reply:
oops!  sorry!  signed the commit..